src/main/scala/is/hail/methods/KinshipMatrix.scala

```diff
@@ -38,18 +39,18 @@ case class KinshipMatrix(hc: HailContext, sampleSignature: Type, matrix: Indexed
 
     val filteredRows = matrix.rows.filter(ir => !sampleIndicesToDropSet(ir.index.toInt))
     val filteredRowsAndCols = filteredRows.map(ir => {
-      val InsertionPoint(numBelowToDelete) = sampleIndicesToDropArray.search(ir.index.toInt)
-      val index = ir.index - numBelowToDelete
+      val InsertionPoint(nBelowToDelete) = sampleIndicesToDropArray.search(ir.index.toInt)
+      val index = ir.index - nBelowToDelete
       val vecArray = ir.vector.toArray
-      val filteredArray = sampleIndicesToTakeArray.map(i => vecArray(i))
-      IndexedRow(index, Vectors.dense(filteredArray))
+      val filteredArray = sampleIndicesToTakeArray.map(vecArray)
+      IndexedRow(index, SparkVectors.dense(filteredArray))
     })
 
     KinshipMatrix(hc, sampleSignature, new IndexedRowMatrix(filteredRowsAndCols), filteredSamplesIds, nVariantsUsed)
   }
   
   def eigen(optNEigs: Option[Int]): Eigendecomposition = {
-    val K = matrix.toLocalMatrix().asBreeze().toDenseMatrix
+    val K = matrix.toLocalMatrix().asBreeze().asInstanceOf[DenseMatrix[Double]]
```

Comment:
I could change asBreeze to additionally cast to Dense or Sparse, so this wouldn't be needed in other places. Thoughts?